File: app/src/test/java/com/immrtldragon/detoxspace/domain/InvitationPolicyTest.kt
package com.immrtldragon.detoxspace.domain

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class InvitationPolicyTest {
    @Test fun sentInvitationCanBeAcceptedDeclinedOrCancelled() {
        assertTrue(InvitationPolicy.canTransition(InvitationState.SENT, InvitationState.ACCEPTED))
        assertTrue(InvitationPolicy.canTransition(InvitationState.SENT, InvitationState.DECLINED))
        assertTrue(InvitationPolicy.canTransition(InvitationState.SENT, InvitationState.CANCELLED))
    }

    @Test fun completedInvitationIsTerminal() {
        assertFalse(InvitationPolicy.canTransition(InvitationState.COMPLETED, InvitationState.SENT))
        assertFalse(InvitationPolicy.canTransition(InvitationState.COMPLETED, InvitationState.CANCELLED))
    }

    @Test fun acceptedInvitationCanBeCompleted() {
        assertTrue(InvitationPolicy.canTransition(InvitationState.ACCEPTED, InvitationState.COMPLETED))
    }
}

File: backend/src/invitation-policy.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { canTransition } from "./invitation-policy.js";

test("sent invitation can be accepted", () => assert.equal(canTransition("SENT", "ACCEPTED"), true));
test("completed invitation is terminal", () => assert.equal(canTransition("COMPLETED", "SENT"), false));
test("recipient cannot move declined invitation", () => assert.equal(canTransition("DECLINED", "ACCEPTED"), false));
